### ISSUE TITLE
Date Exclusion Array Issue

### ISSUE THREAD

**Issue by jsh562** [andruhon/moment-weekday-calc#12]
Whenever I use the date calculations in a loop, if, or another type of method, the array that I use for exclusions seems to always lose its values unless I initialize the array again within that same loop, if, or whatever method that is being used.
I have noticed that when I remove the moment() function from within the loop, if, or other method, then the inclusion array still remembers its values. When I add the moment() function back, it forgets the values unless I re-initialize them within that loop, if, or other method being used.

**WITHOUT moment():**
```

var date_calculation__working_day_array = [1,2,3,4,5];
var date_calculation__date_exclusion_array = ['2013-01-01',
			'2013-05-27',
			'2013-07-04',
			'2013-07-05',
			'2013-09-02'];
var date_calculation__date_inclusion_array = [];

$("#submit").on("click", function(){
   console.log(date_calculation__working_day_array );
   console.log(date_calculation__date_exclusion_array );
});
```
Output:
```
[1, 2, 3, 4, 5]
["2013-01-01", "2013-05-27", "2013-07-04", "2013-07-05", "2013-09-02"]
```
AS EXPECTED.

--------------------------------------------

**WITH moment():**
```

var date_calculation__working_day_array = [1,2,3,4,5];
var date_calculation__date_exclusion_array = ['2013-01-01',
			'2013-05-27',
			'2013-07-04',
			'2013-07-05',
			'2013-09-02'];
var date_calculation__date_inclusion_array = [];

$("#submit").on("click", function(){
   console.log(date_calculation__working_day_array );
   console.log(date_calculation__date_exclusion_array );

   var driven_to_date = moment(date).addWeekdaysFromSet({ 
				'workdays': current_tid__days_from_driver_date, //required, it is also possible to use 'days' alias  
				'weekdays': date_calculation__working_day_array, //optional  
				'exclusions': date_calculation__date_exclusion_array, //optional  
				'inclusions': date_calculation__date_inclusion_array //optional  
			}).format("YYYY-MM-DD");
});
```
Output:
```
[1, 2, 3, 4, 5]
undefined
```
exclusion array is undefined

**Environment:**
Windows 7
Chrome/52.0.2743.116 Safari/537.36
```

**Comment by andruhon** [comment]
This example seems to work for me:

```JavaScript
var date_calculation__working_day_array = [1,2,3,4,5];
var date_calculation__date_exclusion_array = ['2013-01-01',
            '2013-05-27',
            '2013-07-04',
            '2013-07-05',
            '2013-09-02'];
var date_calculation__date_inclusion_array = [];

for (var i = 0; i < 10; i++) {
   console.log(date_calculation__working_day_array );
   console.log(date_calculation__date_exclusion_array );

   var driven_to_date = moment(moment(Math.random()*1000000000000)).addWeekdaysFromSet({ 
                'workdays': Math.ceil(Math.random()*25), //required, it is also possible to use 'days' alias
                'weekdays': date_calculation__working_day_array, //optional
                'exclusions': date_calculation__date_exclusion_array, //optional
                'inclusions': date_calculation__date_inclusion_array //optional
            }).format("YYYY-MM-DD");
}
```
Outputs:
```
[1, 2, 3, 4, 5]
VM267:11 ["2013-01-01", "2013-05-27", "2013-07-04", "2013-07-05", "2013-09-02"]
VM267:10 [1, 2, 3, 4, 5]
VM267:11 ["2013-01-01", "2013-05-27", "2013-07-04", "2013-07-05", "2013-09-02"]
VM267:10 [1, 2, 3, 4, 5]
VM267:11 ["2013-01-01", "2013-05-27", "2013-07-04", "2013-07-05", "2013-09-02"]
VM267:10 [1, 2, 3, 4, 5]
VM267:11 ["2013-01-01", "2013-05-27", "2013-07-04", "2013-07-05", "2013-09-02"]
VM267:10 [1, 2, 3, 4, 5]
VM267:11 ["2013-01-01", "2013-05-27", "2013-07-04", "2013-07-05", "2013-09-02"]
VM267:10 [1, 2, 3, 4, 5]
VM267:11 ["2013-01-01", "2013-05-27", "2013-07-04", "2013-07-05", "2013-09-02"]
VM267:10 [1, 2, 3, 4, 5]
VM267:11 ["2013-01-01", "2013-05-27", "2013-07-04", "2013-07-05", "2013-09-02"]
VM267:10 [1, 2, 3, 4, 5]
VM267:11 ["2013-01-01", "2013-05-27", "2013-07-04", "2013-07-05", "2013-09-02"]
VM267:10 [1, 2, 3, 4, 5]
VM267:11 ["2013-01-01", "2013-05-27", "2013-07-04", "2013-07-05", "2013-09-02"]
VM267:10 [1, 2, 3, 4, 5]
VM267:11 ["2013-01-01", "2013-05-27", "2013-07-04", "2013-07-05", "2013-09-02"]
```

Which version of weekdayCalc do you use? It seems like it was fixed with 1.1.1 for #8

**Comment by jsh562** [comment]
Oh I lost track of that.
Works with 1.1.1, Thanks!
